examples/novatel/rangecmp/range_decompressor_example.cpp

```diff
@@ -135,15 +135,19 @@ int main(int argc, char* argv[])
     stReadData.uiDataSize = sizeof(acFileStreamBuffer);
 
     unsigned char acFrameBuffer[MAX_ASCII_MESSAGE_LENGTH];
-    unsigned char* pucReadBuffer = acFrameBuffer;
+    unsigned char* pucFrameBuffer = acFrameBuffer;
     unsigned char acEncodeBuffer[MAX_ASCII_MESSAGE_LENGTH];
     unsigned char* pucEncodedMessageBuffer = acEncodeBuffer;
 
     InputFileStream clIFS(sInFilename.c_str());
     OutputFileStream clOFS(sInFilename.append(".DECOMPRESSED.").append(sEncodeFormat).c_str());
+    OutputFileStream clUnknownBytesOFS(sInFilename.append(".UNKNOWN").c_str());
     StreamReadStatus stReadStatus;
 
-    STATUS eStatus = STATUS::UNKNOWN;
+    STATUS eFramerStatus = STATUS::UNKNOWN;     // Framer Status
+    STATUS eDecompressStatus = STATUS::UNKNOWN; // Decompressor Status
```

Comment:
Is there an advantage to using separate variables for each stage?

Reply:
moreso, it is consistent with the converter components example. Also, there was an issue where the status for one stage was being overwritten in another. It led to a case where there was code that was impossible to run. I've separated them for those reasons basically.

Reply:
ok, in that case use auto here as the type is easily deduced on the same line